bin/spack

```diff
@@ -10,9 +10,9 @@
 # Following line is a shell no-op, and starts a multi-line Python comment.
 # See https://stackoverflow.com/a/47886254
 """:"
-# prefer python3, then python, then python2
-for cmd in python3 python python2; do
-   command -v > /dev/null $cmd && exec $cmd $0 "$@"
+# prefer SPACK_PYTHON environment variable, python3, python, then python2
+for cmd in "$SPACK_PYTHON" python3 python python2; do
```

Comment:
```suggestion
for cmd in $SPACK_PYTHON python3 python python2; do
```
The difference is that if `SPACK_PYTHON` isn't set, we loop over 3 commands instead of 4 ("" python3 python python2)

Reply:
I considered that but wanted to be able to allow spaces in the path so the quotes are needed.  In the case of an empty string `command -v > /dev/null ""` will fail and iteration will proceed to `python3`.  However, with the addition of the `setup-env.*sh` changes the use case for an empty string will me much less common.